### PR TITLE
Bump Iceberg to 1.4.0, support Spark 3.5

### PR DESCRIPTION
**Pull request by snazy**

### CODE REVIEW

File: build-logic/src/main/kotlin/Utilities.kt [hunk 1]
```diff
@@ -354,7 +354,8 @@ fun javaVersionForSpark(sparkMajorVersion: String): Int {
     "3.1",
     "3.2" -> 11
     "3.3",
-    "3.4" ->
+    "3.4",
```
Comment (ajantha-bhat) [inline build-logic/src/main/kotlin/Utilities.kt:357]: 3.1 can be removed in line 354 and javadoc can be updated.

File: gradle/libs.versions.toml [hunk 2]
```diff
@@ -80,13 +80,17 @@ iceberg-spark-extensions33-scala212 = { module = "org.apache.iceberg:iceberg-spa
 iceberg-spark-extensions33-scala213 = { module = "org.apache.iceberg:iceberg-spark-extensions-3.3_2.13", version.ref = "iceberg" }
 iceberg-spark-extensions34-scala212 = { module = "org.apache.iceberg:iceberg-spark-extensions-3.4_2.12", version.ref = "iceberg" }
 iceberg-spark-extensions34-scala213 = { module = "org.apache.iceberg:iceberg-spark-extensions-3.4_2.13", version.ref = "iceberg" }
+iceberg-spark-extensions35-scala212 = { module = "org.apache.iceberg:iceberg-spark-extensions-3.5_2.12", version.ref = "iceberg" }
```
Comment (ajantha-bhat) [inline gradle/libs.versions.toml:83]: 3.1 can be removed from line 76

File: integrations/spark-extensions/v3.5/src/intTest/java/org/projectnessie/spark/extensions/ITNessieStatements.java [hunk 3]
```diff
@@ -0,0 +1,18 @@
+/*
+ * Copyright (C) 2020 Dremio
```
Comment (ajantha-bhat) [inline integrations/spark-extensions/v3.5/src/intTest/java/org/projectnessie/spark/extensions/ITNessieStatements.java:2]: nit: 2023

File: integrations/spark-scala.properties [hunk 4]
```diff
@@ -19,9 +19,10 @@
 # The first Scala version for each Spark version is the one that will
 # appear as a project in IntelliJ.
 
-sparkVersions=3.1,3.2,3.3,3.4
+sparkVersions=3.2,3.3,3.4,3.5
 
 sparkVersion-3.1-scalaVersions=2.12
```
Comment (ajantha-bhat) [inline integrations/spark-scala.properties:24]: we can look up "3.1" in the code. Still documentation and some infra files like this has reference to it.

**Review by ajantha-bhat — APPROVED** [review]

**Review by adutra — APPROVED** [review]

**Review by ajantha-bhat — APPROVED** [review]

**Pull request merged**